src/geometamaker/config.py

```diff
@@ -0,0 +1,60 @@
+import logging
+import os
+
+import platformdirs
+
+from . import models
+
+
+LOGGER = logging.getLogger(__name__)
+
+DEFAULT_CONFIG_PATH = os.path.join(
+    platformdirs.user_config_dir(), 'geometamaker_profile.yml')
+
+
+class Config(object):
+    """Encapsulates user-settings such as a metadata Profile."""
+
+    def __init__(self, config_path=DEFAULT_CONFIG_PATH):
+        """Load a Profile from a config file.
+
+        Use a default user profile if none given.
+
+        Args:
+            config_path (str): path to a local yaml file
+
+        """
+        self.config_path = config_path
+        self.profile = models.Profile()
+
+        try:
+            self.profile = models.Profile.load(self.config_path)
+        except FileNotFoundError as err:
+            LOGGER.debug(err)
+            pass
+        # an invalid profile should raise a TypeError
+        except TypeError as err:
+            LOGGER.warning(err)
+            LOGGER.warning(
+                f'{self.config_path} contains an inavlid profile. '
+                'It will be ignored. You may wish to delete() it.')
+
+    def __repr__(self):
+        """Represent config as a string."""
+        return (
+            f'{self.config_path}: \n'
+            f'{self.profile}')
+
+    def save(self, profile):
+        """Save a Profile to a local config file.
+
+        Args:
+            profile (geometamaker.models.Profile)
+        """
+        LOGGER.info(f'writing profile to {self.config_path}')
+        profile.write(self.config_path)
```

Comment:
This method is interesting because it is tightly coupled to the functionality of `Profile`, yet is a member method here of this class.  If we extend the `Config` class later on to cover more than just `Profile`s, how would this method need to change?

Reply:
Great question. I struggled a bit with this tight coupling between `Profile` and `Config` and ended up with two classes so that the `Config` class could handle knowing the default location for a config file. And in case there is other config info to store later on.

I could imagine re-structuring the config file so that is has a section for a `profile`, rather than assuming the entire file is the `profile`. Then the `Config.__init__` and `save` methods would change a bit. I don't know if it's smart to do all that now, or wait until we want to extend the config data somehow.

Reply:
Although I do think we'll want to update it eventually, I think it's completely fine to wait until we have other information in the config object that isn't in the Profile.